.github/workflows/swift-toolchain.yml

```diff
@@ -485,17 +497,17 @@ jobs:
 
       - uses: actions/upload-artifact@v4
         with:
-          name: build-tools
+          name: build-tools-${{ matrix.os }}
           path: |
-            ${{ github.workspace }}/BinaryCache/0/bin/llvm-tblgen.exe
-            ${{ github.workspace }}/BinaryCache/0/bin/clang-tblgen.exe
-            ${{ github.workspace }}/BinaryCache/0/bin/lldb-tblgen.exe
-            ${{ github.workspace }}/BinaryCache/0/bin/llvm-config.exe
-            ${{ github.workspace }}/BinaryCache/0/bin/clang-pseudo-gen.exe
-            ${{ github.workspace }}/BinaryCache/0/bin/clang-tidy-confusable-chars-gen.exe
-            ${{ github.workspace }}/BinaryCache/0/bin/swift-def-to-strings-converter.exe
-            ${{ github.workspace }}/BinaryCache/0/bin/swift-serialize-diagnostics.exe
-            ${{ github.workspace }}/BinaryCache/0/bin/swift-compatibility-symbols.exe
+            ${{ github.workspace }}/BinaryCache/0/bin/llvm-tblgen${{ inputs.exe_suffix }}
+            ${{ github.workspace }}/BinaryCache/0/bin/clang-tblgen${{ inputs.exe_suffix }}
+            ${{ github.workspace }}/BinaryCache/0/bin/lldb-tblgen${{ inputs.exe_suffix }}
+            ${{ github.workspace }}/BinaryCache/0/bin/llvm-config${{ inputs.exe_suffix }}
+            ${{ github.workspace }}/BinaryCache/0/bin/clang-pseudo-gen${{ inputs.exe_suffix }}
+            ${{ github.workspace }}/BinaryCache/0/bin/clang-tidy-confusable-chars-gen${{ inputs.exe_suffix }}
+            ${{ github.workspace }}/BinaryCache/0/bin/swift-def-to-strings-converter${{ inputs.exe_suffix }}
+            ${{ github.workspace }}/BinaryCache/0/bin/swift-serialize-diagnostics${{ inputs.exe_suffix }}
+            ${{ github.workspace }}/BinaryCache/0/bin/swift-compatibility-symbols${{ inputs.exe_suffix }}
```

Comment:
Can we not use a pattern instead of the exe suffix?

Reply:
What do you mean by "using a pattern"?

Reply:
I'm guessing regex-like path pattern matching? I don't know if this works but maybe something like `{{ github.workspace }}/BinaryCache/0/bin/llvm-tblgen(.exe)` (where `.exe` may or may not exist) might work? This may be relevant: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

Reply:
This action uses this particular glob library: https://github.com/actions/toolkit/blob/main/packages/glob/README.md

Unfortunately, it does not support `(optional-pattern)` and only supports `?` as "any single character", not as "0 or 1 of the previous character", so neither `(.exe)` nor `.?e?x?e?` worked. I am open to other suggestions.

Reply:
I wonder if `.*` as in `../llvm-tblgen.*` would be. I think this should be fine as long as there isn't a path that is a subpart of another (likely the case here)? If this doesn't pan out, I suggest sticking to what you had originally.

Reply:
I thought about it but my concern is over-packaging. We may end up with a bunch of large unrelated files like PDBs.

Reply:
Ah, that's fair. Unfortunate that we cannot simply glob.

Reply:
A good point about the pdb and such

Reply:
I moved to exporting the names as a sub-step. We'll have to change the consumption targets down the line but this can wait when I get there.